### PR TITLE
Feature settle up implementation

### PR DESCRIPTION
**Pull request by HarishChoure**
Linked Work Items

https://expense-management-system-batch-1-2022.atlassian.net/browse/EMSB12-79
Checklist
 

**Checklist**
- [x] Code adheres to **[Git and Pull Request Guidelines]**
- [x] Code adheres to **[Angular Guidelines]**
- [x] Code adheres to **[Front-end Checklist]**

### CODE REVIEW

File: src/app/expense/expense-list/expense-list.component.ts [hunk 1]
```diff
@@ -50,19 +50,18 @@ export class ExpenseListComponent implements OnInit {
...
    */
   getExpenseList(): void{
-
     this.expenseService.getExpenseList().subscribe(async data => {
       this.expense=data;
       this.expenseItem = this.expense.items;
       for(let i=0;i<this.expenseItem.length;i++){
         this.groupService.getGroupDetailByGroupId(this.expenseItem[i].groupId).subscribe( data => {
           this.expenseItem[i].groupName= data.name
+          console.log(data.name)
```
Comment (AshaPatel216) [inline src/app/expense/expense-list/expense-list.component.ts]: Remove console log

File: src/app/services/settle-up.service.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { Observable } from 'rxjs';
+@Injectable({
+  providedIn: 'root'
+})
+export class SettleUpService {
+  /**
+   * constructor
+   */
+  constructor(private http:HttpClient) {}
+  private baseUrl = 'https://localhost:44329';
```
Comment (AshaPatel216) [inline src/app/services/settle-up.service.ts]: Variable must be declare before constructor. Missing type.

File: src/app/expense/expense-list/expense-list.component.ts (hunk 1 above)
Comment (HarishChoure) [reply src/app/expense/expense-list/expense-list.component.ts]: Ok Di

File: src/app/settle-up/settle-up.component.ts [hunk 3]
```diff
@@ -1,12 +1,54 @@
...
 
-  currency: String[] = ['MYR', 'SGD', 'USD'];
-  groups: String[] = ['Mathura', 'Vanaras', 'Goa'];
+  /**
+   * getPaymentID are used to take payment Id from the user
+   */
+  getPaymentID(event: string) {
+    this.paymentID = event
+    this.paymentMethod(this.paymentID)
+  }
```
Comment (AshaPatel216) [inline src/app/settle-up/settle-up.component.ts:54]: Missing retun type. Add void if no return type. check whole PR

File: src/app/settle-up/settle-up.component.ts [hunk 4]
```diff
@@ -1,12 +1,54 @@
...
+    },
+    )
+  }
 
-  currency: String[] = ['MYR', 'SGD', 'USD'];
-  groups: String[] = ['Mathura', 'Vanaras', 'Goa'];
+  /**
+   * getPaymentID are used to take payment Id from the user
+   */
+  getPaymentID(event: string) {
```
Comment (AshaPatel216) [inline src/app/settle-up/settle-up.component.ts]: @params missing

File: src/app/settle-up/settle-up.component.ts (hunk 3 above)
Comment (HarishChoure) [reply src/app/settle-up/settle-up.component.ts:54]: Ok I have added void in the getPaymentId method

File: src/app/settle-up/settle-up.component.ts (hunk 4 above)
Comment (HarishChoure) [reply src/app/settle-up/settle-up.component.ts]: Ok Di I have added params in the comment.

**Review by AshaPatel216 — APPROVED** [review]

**Pull request merged**